rest-api/dao/biobank_order_dao.py

```diff
@@ -214,11 +215,21 @@ def _set_participant_summary_fields(self, obj, participant_summary):
     participant_summary.biospecimenProcessedSiteId = obj.processedSiteId
     participant_summary.biospecimenFinalizedSiteId = obj.finalizedSiteId
     participant_summary.lastModified = clock.CLOCK.now()
+
+    has_dna_test = False
     for sample in obj.samples:
+      if sample.test in BIOBANK_DNA_TESTS:
+        has_dna_test = True
       status_field = 'sampleOrderStatus' + sample.test
       status, time = self._get_order_status_and_time(sample, obj)
       setattr(participant_summary, status_field, status)
       setattr(participant_summary, status_field + 'Time', time)
+    if has_dna_test:
```

Comment:
The logic for being a core participant is more complicated than this... it's not just having DNA samples stored / ordered, it's also having completed EHR consent, physical measurements, and baseline questionnaires. See:

https://github.com/all-of-us/raw-data-repository/blob/master/rest-api/dao/participant_summary_dao.py#L45

(for SQL run from biobank samples pipeline) and

https://github.com/all-of-us/raw-data-repository/blob/master/rest-api/dao/participant_summary_dao.py#L327

for Python logic run when updating the participant summary. This is what needs to be updated.